code/modules/mob/living/carbon/slime/slime_extractor.dm

```diff
@@ -0,0 +1,121 @@
+	icon = 'icons/obj/stationobjs.dmi'
+	icon_state = "slime_extractor"
+	density = TRUE
+	anchored = TRUE
+	use_power = TRUE
+	var/slime_limit = 2 // how many slimes we can process at a time
+	var/extraction_speed = 150
+	idle_power_usage = 15
+	active_power_usage = 50
+	var/list/mob/living/carbon/slime/extract_slimes = list()
+
+	component_types = list(
+		/obj/item/circuitboard/slime_extractor,
+		/obj/item/stock_parts/matter_bin,
+		/obj/item/stock_parts/micro_laser,
+		/obj/item/stack/cable_coil{amount = 5}
+	)
+
+/obj/machinery/slime_extractor/examine(mob/user)
+	. = ..()
+	to_chat(user, FONT_SMALL(SPAN_NOTICE("It can hold <b>[slime_limit] slime\s</b> at a time.")))
+	if(length(extract_slimes))
+		to_chat(user, FONT_SMALL(SPAN_WARNING("It is currently processing <b>[length(extract_slimes)] slime\s</b>.")))
+
+/obj/machinery/slime_extractor/update_icon()
+	cut_overlays()
+	if(panel_open)
+		var/mutable_appearance/panel_overlay = mutable_appearance(icon, "[icon_state]-panel")
+		add_overlay(panel_overlay)
+	if(length(extract_slimes))
+		var/mutable_appearance/interior_overlay = mutable_appearance(icon, "[icon_state]-interior", EFFECTS_ABOVE_LIGHTING_LAYER)
+		add_overlay(interior_overlay)
+		var/mutable_appearance/spinning_overlay = mutable_appearance(icon, "[icon_state]-running", EFFECTS_ABOVE_LIGHTING_LAYER)
+		add_overlay(spinning_overlay)
+		set_light(2.5, 1, COLOR_VIOLET)
+	else
+		set_light(FALSE)
+
+/obj/machinery/slime_extractor/RefreshParts()
+	..()
+	slime_limit = initial(slime_limit)
+	extraction_speed = initial(extraction_speed)
+
+	for(var/obj/item/stock_parts/P in component_parts)
+		if(ismatterbin(P))
+			slime_limit *= P.rating
+		else if(ismicrolaser(P))
+			extraction_speed = extraction_speed / P.rating
+
+/obj/machinery/slime_extractor/attackby(obj/item/O, mob/user)
+	if(length(extract_slimes))
+		to_chat(user, SPAN_WARNING("You can't modify \the [src] while it's busy. Please wait for the completion of previous operation."))
+		return
+
+	if(default_deconstruction_screwdriver(user, O))
+		return
+	if(default_deconstruction_crowbar(user, O))
+		return
+	if(default_part_replacement(user, O))
+		return
+
+/obj/machinery/slime_extractor/MouseDrop_T(atom/dropping, mob/user)
+	if(!Adjacent(user))
+		to_chat(user, SPAN_WARNING("You can't reach \the [src]!"))
+		return
+	if(istype(dropping, /mob/living/carbon/slime))
+		var/mob/living/carbon/slime/slimey = dropping
+		if(length(extract_slimes) >= slime_limit)
+			to_chat(user, SPAN_WARNING("\The [src] is fully loaded!"))
+			return
+		user.visible_message(SPAN_NOTICE("\The [user] starts loading \the [slimey] into \the [src]..."), SPAN_NOTICE("You start loading \the [slimey] into \the [src]..."))
+		if(do_after(user, 20))
+			if(!Adjacent(user))
+				to_chat(user, SPAN_WARNING("You can't reach \the [src]!"))
+				return
```

Comment:
do_after already checks if we moved, and we already checked if they are Adjacent at the start, so do we care if they are still Adjacent? I think no other machine lets you move while using it, even if you remain Adjacent so we should keep the same behaviour here.